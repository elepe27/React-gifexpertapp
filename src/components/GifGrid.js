import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

    

    const {data:imgs, loading} = useFetchGifs(category);
    
    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>

            {loading && <p className='ard animate__animated animate__flash'>Loading</p>}
            
            <div className='card-grid'>

                {imgs.map(img =>
                    <GifGridItem
                        key={img.id}
                        {...img}
                    />
                )}

            </div>
        </>
    )
}

GifGrid.propTypes ={
    category: PropTypes.string.isRequired
}
