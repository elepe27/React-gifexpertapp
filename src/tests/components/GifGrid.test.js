import {shallow} from 'enzyme'
import '@testing-library/jest-dom'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componente GifGrid', () => {

    const category = 'One Piece';
    
    
    test('Debe de mostrar el componente', () => {
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });

        const wrapper = shallow(<GifGrid category ={category}/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {


        const gifs =[{
            id:'abc',
            url:'localhost:2020',
            title: 'Prueba'
        },
        {
            id:'abc',
            url:'localhost:2020',
            title: 'Prueba'
        }];
            

        

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });

        
        const wrapper = shallow(<GifGrid category ={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        

        

    });
    
    
    
})
