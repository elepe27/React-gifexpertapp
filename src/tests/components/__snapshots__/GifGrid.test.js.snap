// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe de mostrar el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Piece
  </h3>
  <p
    className="ard animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente GifGrid Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Piece
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Prueba"
      url="localhost:2020"
    />
    <GifGridItem
      id="abc"
      key="abc"
      title="Prueba"
      url="localhost:2020"
    />
  </div>
</Fragment>
`;
